spec/models/bus_stops_route_spec.rb

```diff
@@ -35,4 +35,74 @@
       end
     end
   end
+  describe 'import' do
+    let(:main_variant) { %w[A B C D E] }
+    let(:route_dir) { %w[31 North] }
+    context 'all stops are in longest variant' do
+      it 'sequences stops correctly' do
+        stop_hash = { route_dir => { 'm_North' => main_variant } }
+        BusStopsRoute.import(stop_hash)
+        expect(stop_hash[route_dir]).to be main_variant
+      end
+    end
+    context 'stop (but not first stop) is not in longest variant' do
+      context 'one stop is in other variant' do
+        it 'sequences stops correctly' do
+          sequenced_stops = %w[A F B C D E]
+          other_variant = %w[A F E]
+          stop_hash = { route_dir =>
+                        { 'm_North' => main_variant,
+                          'o_North' => other_variant } }
+          BusStopsRoute.import(stop_hash)
+          expect(stop_hash[route_dir]).to eql sequenced_stops
+        end
+      end
+      context 'multiple stops are in other variant' do
+        it 'sequences stops correctly' do
+          sequenced_stops = %w[A F G H B C D E]
+          other_variant = %w[A F G H E]
+          stop_hash = { route_dir =>
+                        { 'm_North' => main_variant,
+                          'o_North' => other_variant } }
+          BusStopsRoute.import(stop_hash)
+          expect(stop_hash[route_dir]).to eql sequenced_stops
+        end
+      end
+    end
+    context 'first stop is not in longest variant' do
+      context 'one stop is in other variant' do
+        it 'sequences stops correctly' do
+          sequenced_stops = %w[A B C F D E]
```

Comment:
I know this pattern repeats itself. Should I -- and if so, how do I -- abstract it out?

Reply:
Ah, okay. I was having trouble with the `doesnt_matter` parts. Hooray!

Reply:
Oh, got it